lib/grpc/adapter/client.ex

```diff
@@ -0,0 +1,27 @@
+defmodule GRPC.ClientAdapter do
+  @moduledoc false
+
+  alias GRPC.Client.Stream
+  alias GRPC.Channel
+
+  @type opts :: any
```

Comment:
I think it's better to use `opts :: keyword()` instead (perhaps even avoiding a custom type, inlining this wherever needed).

This would warrant some changes to the current implementation, but provides a better interface